functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -6,71 +10,124 @@
+        logging.debug('Executed the query')          
+        row = cursor.fetchone()
+        logging.info(f"Got result: {row}")
+        if not row:
+            logging.error('No record with the requested parameters')
+            return func.HttpResponse('Task not found', status_code=500)
+        else:
+            columns = [column[0] for column in cursor.description]
+            data = dict(zip(columns, row))
+        return func.HttpResponse(json.dumps(data, default=str), status_code=200, mimetype="application/json")
     finally:
         cursor.close()
         cnxn.close()
-        logging.info('Closed the connection')
+        logging.info('Closed the db connection')   
 
-#POST API method function       
-def post(param):
-    return ('You selected POST method with {param} values. Functionality under construction')
-            # STARTER CODE FOR NEXT SPRINT, SQL QUERY TESTED
-            #     #insert row into tasks table / create a new task
-            #     #createdDate #'20120618 10:34:09 AM' and title/description are hardcoded for sprint1, update with automated date stamp and url params later
-            #     sql_query = ("""INSERT INTO dbo.tasks (userId, title, description, createdDate)
-            #     VALUES (?, ?, ?, '20120618 10:34:09 AM')""")
-            #     cursor.execute(sql_query, userId, 'Do It', 'Almost like Nike motto')   
-            #     logging.info('Executed the query')
-
-#UPDATE API method function
-def update(param1, param2):
+#PUT API method function
+def update(param1, param2, task_req_body):
     #connects to db
     cnxn = connect()
     cursor = cnxn.cursor()
-    logging.info('opened connection')        
-    logging.info('Going to execute update query')
     try:
-        # update task: title and description. Client passes userId and taskId, 
-        # for sprint 1 other fields are HARDCODED
-        sql_query = ("""UPDATE [dbo].[tasks]
-        SET title = 'Title updated by team1', description = 'Description updated by team1'
-        WHERE userId = ? AND taskId = ?""")   
-        rowcount = cursor.execute(sql_query, param1, param2).rowcount
-        logging.info(f"Executed the query: {rowcount} rows affected")
-#commits changes to db
+        logging.info('opened connection')        
+        logging.info('Going to execute UPDATE task query')
+        # task_req_body input check
+        logging.debug("Checking the request body for necessary fields to update a task")
+        try:
+            assert 'title' in task_req_body, "New task request body did not contain field: 'title'"
+            assert 'description' in task_req_body, "New task request body did not contain field: 'description'"
+            assert 'dueDate' in task_req_body, "New user request body did not contain field: 'dueDate'"
+        except AssertionError as task_req_body_content_error:
+            logging.critical('New user request body did not contain fields to update a task')

```

Comment:
While I agree that db connection issues are more likely than not supposed to be critical, I don't think request errors are. We should probably ask Toddy tonight, but I really think that if it doesn't require changes on our side (i.e. they just need to change the request body), then it's more on the side of an error rather than a critical.

Reply:
Sure, if we draw the parallel with 500 vs. 404 discussion, debug would make sense here vs. critical

Reply:
We can clarify this tonight with toddy, but I would agree with Lenny in this case.